provider/oci/common/client.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package common
+
+import (
+	// "context"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	"github.com/juju/errors"
+	// "github.com/juju/juju/instance"
+	// "github.com/juju/juju/network"
+
+	ociCommon "github.com/oracle/oci-go-sdk/common"
+	ociCore "github.com/oracle/oci-go-sdk/core"
+	ociIdentity "github.com/oracle/oci-go-sdk/identity"
+)
+
+type JujuConfigProvider struct {
+	Key         []byte
+	Fingerprint string
+	Passphrase  string
+	Tenancy     string
+	User        string
+	OCIRegion   string
+}
+
+type ociClient struct {
+	ociCore.ComputeClient
+	ociCore.BlockstorageClient
+	ociCore.VirtualNetworkClient
+	ociIdentity.IdentityClient
+
+	ociCommon.ConfigurationProvider
+}
+
+func NewOCIComputeClient(provider ociCommon.ConfigurationProvider) (OCIComputeClient, error) {
```

Comment:
Are these functions worth having? They seem like they just call the back object, which already implements the interface that we want to use.
Maybe having them all grouped here is still useful.

Reply:
yes, please do.